ingest/config/validate/validate.go

```diff
@@ -31,8 +31,19 @@ var (
 // The config handling function will take those two paths and process a config and return two values.
 // The first value is an opaque object (hence the reflect voodoo).  The second value is an error object
 // the point of this function is to make it easy for ingester writers to just hand in their GetConfig function
-// and the two paths and get a "go/no go" on the configurations
+// and the two paths and get a "go/no go" on the configurations.
+// ValidateConfig also asserts that the provided config object can return an IngestBaseConfig object.
 func ValidateConfig(fnc interface{}, pth, confdPath string) {
+	validateConfig(fnc, pth, confdPath, true) // this is used by ingesters
+}
+
+// ValidateConfigBase behaves the same as ValidateConfig but does not assert that the provided config
```

Comment:
It seems kind of confusing that "ValidateConfig**Base**" is the one that *doesn't* assert that the config can return an Ingest**Base**Config object...

Reply:
I chose this because it has less assertions.  The other one asserts it's an ingester config